src/stream/src/common/table/state_table.rs

```diff
@@ -660,12 +673,28 @@ impl<S: StateStore> StateTable<S> {
         &mut self,
         buffer: BTreeMap<Vec<u8>, RowOp>,
         epoch: u64,
+        watermark: Option<&ScalarImpl>,
```

Comment:
This is a common change as seen in https://github.com/risingwavelabs/risingwave/pull/6330. Can we make a PR just focused on the state table change and add some unit tests for the changes in `test_streaming_table.rs`?

Reply:
I think it'll make it easier to review the changes in one place.

Reply:
> This is a common change as seen in #6330. Can we make a PR just focused on the state table change and add some unit tests for the changes in `test_streaming_table.rs`?

just created an independent PR #6495 for this. cc @Little-Wallace